generators/chipyard/src/main/scala/ConfigMixins.scala

```diff
@@ -35,117 +35,40 @@ import ConfigValName._
 // Common Parameter Mixins
 // -----------------------
 
-/**
- * Mixin to add the Chipyard bootrom
- */
 class WithBootROM extends Config((site, here, up) => {
   case BootROMParams => BootROMParams(
     contentFileName = s"./bootrom/bootrom.rv${site(XLen)}.img")
 })
 
 // DOC include start: gpio mixin
-/**
- * Mixin to add GPIOs and tie them off outside the DUT
- */
 class WithGPIO extends Config((site, here, up) => {
   case PeripheryGPIOKey => Seq(
     GPIOParams(address = 0x10012000, width = 4, includeIOF = false))
-  case BuildTop => (clock: Clock, reset: Bool, p: Parameters, success: Bool) => {
-    val top = up(BuildTop, site)(clock, reset, p, success)
-    // TODO: Currently FIRRTL will error if the GPIO input
-    // pins are unconnected, so tie them to 0.
-    // In future IO cell blackboxes will replace this with
-    // more correct functionality
-    for (gpio <- top.gpio) {
-      for (pin <- gpio.pins) {
-        pin.i.ival := false.B
-      }
-    }
-    top
-  }
 })
 // DOC include end: gpio mixin
 
-/**
- * Mixin to add in UART
- */
 class WithUART extends Config((site, here, up) => {
   case PeripheryUARTKey => Seq(
     UARTParams(address = 0x54000000L, nTxEntries = 256, nRxEntries = 256))
 })
 
-/**
- * Mixin to remove any GPIOs
- */
 class WithNoGPIO extends Config((site, here, up) => {
```

Comment:
Can we explain somewhere in the docs the reason for the `WithNoGPIO` (as opposed to everything else which is an additive). I know the reason, I just think it'll be nice to explain it in the docs as well

Reply:
There seems to be no good place to include this, since there aren't docs on GPIO anywhere. In any case this should get fixed eventually, so its not high priority.